### PR TITLE
Add support for Categories 

### PR DESCRIPTION
**Pull request by stratus-ss**
This PR adds support for adding the following:

New Categories
Items can now be added with a category pre-defined

**Code additions:**

* added self._category_id 
* ATTR_ITEM_CATEGORY
* REGEX_CATEGORY_ID

**Function addtions:**
* get_category_items
* create_category

**Code Modifications:**
*  self._team_id = re.findall(REGEX_TEAM_ID, responseText)[0]
* add_item_to_list now accepts additional argument
* add_item_to_list uses additional argument in payload

You can test the code with something like this

```
import asyncio
from ourgroceries import OurGroceries
 
username = "username"
password = "password"
og = OurGroceries(username, password)
asyncio.run(og.login())
my_list_id = "bK8DWHrHc5pdBBtRdS8q8f"
 
 
def refresh_grocery_lists():
    my_full_list = asyncio.run(og.get_list_items(list_id=my_list_id))
    all_categories = asyncio.run(og.get_category_items())
    return(my_full_list, all_categories)
 
 
def return_category_id(category_to_search_for, all_categories):
    category_to_search_for_lower = category_to_search_for.lower()
    category_id = None
    if len(all_categories['list']['items']) is not 0:
        for category_heading in all_categories['list']['items']:
            # Split the heading because if there is already a duplicate it
            # presents as "{{item}} (2)"
            category_heading_lowered = category_heading['value'].lower().split()[0]
            if category_to_search_for_lower == category_heading_lowered:
                category_id = category_heading['id']
                break
    return(category_id)
 
 
def add_to_my_list(my_full_list, my_list_id, item_name, all_categories, category="uncategorized"):
    # check to make sure the object doesn't exist
    # The groceries live in my_full_list['list']['items']
    # Start with the assumption that the food does not exist
    food_exists = False
    toggle_crossed_off = False
    category_lowered = category.lower()
    for food_item in my_full_list['list']['items']:
        if item_name in food_item['value']:
            print("Already exists!!!")
            try:
                if food_item['crossedOff']:
                    existing_item_id = food_item['id']
                    toggle_crossed_off = True
            except KeyError:
                pass
            food_exists = True
    if not food_exists:
        category_id = return_category_id(category_lowered, all_categories)
        asyncio.run(og.add_item_to_list(my_list_id, item_name, category_id))
        print("Added item")
    else:
        if toggle_crossed_off:
            asyncio.run(og.toggle_item_crossed_off(my_list_id, existing_item_id, cross_off=False))
 
 
def add_category(category_name, all_categories):
    category_id = return_category_id(category_name, all_categories)
    if category_id is None:
        asyncio.run(og.create_category(category_name))
        print("Added Category")
    else:
        print("Category already exists")
 
my_full_list, current_categories = refresh_grocery_lists()
 
add_category("Funnies", current_categories)
my_full_list, current_categories = refresh_grocery_lists()
add_to_my_list(my_full_list, my_list_id, "commics", current_categories, category="Funnies")
```

I am working on a more well rounded application for your library here:
https://github.com/stratus-ss/mycroft-ourgroceries-skill/blob/master/talk_to_ourgroceries.py

### CODE REVIEW

File: ourgroceries/__init__.py [hunk 1]
```diff
@@ -133,6 +137,16 @@ async def create_list(self, name, list_type='SHOPPING'):
...
         return await self._post(ACTION_LIST_CREATE, other_payload)
 
+    async def create_category(self, name):
+        """Create a new category."""
+        _LOGGER.debug('ourgroceries create_list')
+        other_payload = {
+            ATTR_ITEM_VALUE: name,
+            ATTR_LIST_ID: self._category_id,
+        }
+        print(other_payload)
```
Comment (ljmerza) [inline ourgroceries/__init__.py:147]: remove print

File: ourgroceries/__init__.py [hunk 2]
```diff
@@ -105,23 +105,27 @@ async def _get_team_id(self):
         async with aiohttp.ClientSession(cookies=cookies) as session:
             async with session.get(YOUR_LISTS) as resp:
                 responseText = await resp.text()
-                for team_id in re.findall(REGEX_TEAM_ID, responseText):
-                    self._team_id = team_id
-                    _LOGGER.debug('ourgroceries found team_id {}'.format(self._team_id))
+                self._team_id = re.findall(REGEX_TEAM_ID, responseText)[0]
+                self._category_id = re.findall(REGEX_CATEGORY_ID, responseText)[0]
+
```
Comment (ljmerza) [inline ourgroceries/__init__.py:110]: add debug log `_LOGGER.debug('ourgroceries found team_id {}'.format(self._team_id))`

File: ourgroceries/__init__.py [hunk 3]
```diff
@@ -133,6 +137,16 @@ async def create_list(self, name, list_type='SHOPPING'):
         }
         return await self._post(ACTION_LIST_CREATE, other_payload)
 
+    async def create_category(self, name):
+        """Create a new category."""
+        _LOGGER.debug('ourgroceries create_list')
```
Comment (ljmerza) [inline ourgroceries/__init__.py:142]: change `_LOGGER.debug('ourgroceries create_list')` to `_LOGGER.debug('ourgroceries create_category')`

File: ourgroceries/__init__.py [hunk 4]
```diff
@@ -105,23 +105,27 @@ async def _get_team_id(self):
...
+
 
     async def get_my_lists(self):
         """Get our grocery lists."""
         _LOGGER.debug('ourgroceries get_my_lists')
         return await self._post(ACTION_GET_LISTS)
 
+    async def get_category_items(self):
+        """Get category items."""
+        _LOGGER.debug('ourgroceries get_list_items')
```
Comment (ljmerza) [inline ourgroceries/__init__.py:119]: change `_LOGGER.debug('ourgroceries get_list_items')` to `_LOGGER.debug('ourgroceries get_category_items')`

File: ourgroceries/__init__.py [hunk 5]
```diff
@@ -105,23 +105,27 @@ async def _get_team_id(self):
         async with aiohttp.ClientSession(cookies=cookies) as session:
             async with session.get(YOUR_LISTS) as resp:
                 responseText = await resp.text()
-                for team_id in re.findall(REGEX_TEAM_ID, responseText):
-                    self._team_id = team_id
-                    _LOGGER.debug('ourgroceries found team_id {}'.format(self._team_id))
+                self._team_id = re.findall(REGEX_TEAM_ID, responseText)[0]
+                self._category_id = re.findall(REGEX_CATEGORY_ID, responseText)[0]
```
Comment (ljmerza) [inline ourgroceries/__init__.py:109]: leave the `_LOGGER.debug('ourgroceries found team_id {}'.format(self._team_id))` and add one for the category

**Comment by stratus-ss** [comment]
Thanks for the pointers. I made some sloppy logging errors. Good catch

**Pull request merged**